agent/xds/testdata/rbac/empty-top-level-jwt-with-one-permission--httpfilter.golden

```diff
@@ -0,0 +1,53 @@
+{
+  "name": "envoy.filters.http.rbac",
+  "typedConfig": {
+    "@type": "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC",
+    "rules": {
+      "policies": {
+        "consul-intentions-layer7-0": {
+          "permissions": [
+            {
+              "urlPath": {
+                "path": {
+                  "prefix": "some-path"
+                }
+              }
+            }
+          ],
+          "principals": [
+            {
+              "authenticated": {
+                "principalName": {
+                  "safeRegex": {
+                    "googleRe2": {},
+                    "regex": "^spiffe://test.consul/ns/default/dc/[^/]+/svc/web$"
+                  }
+                }
+              }
+            },
+            {
+              "orIds": {
+                "ids": [
+                  { 
+                    "metadata": {
+                      "filter":"envoy.filters.http.jwt_authn",
+                      "path": [
+                        {"key": "jwt_payload_okta_0"},
+                        {"key": "roles"}
+                      ],
+                      "value": {
+                        "stringMatch": {
+                          "exact": "testing"
+                        }
+                      }
+                    }
+                  }
+                ]
+              }
```

Comment:
Do we need to "and" these together?

Reply:
We do, because [principals are matched with OR semantics](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto#config-rbac-v3-policy). This needs to look like: 

```json
{
  "authenticated": {
    "and_ids": {
      "ids": [
        {
          "principalName": {
            "safeRegex": {
              "googleRe2": {},
              "regex": "^spiffe://test.consul/ns/default/dc/[^/]+/svc/web$"
            }
          }
        },
        { 
          "metadata": {
            "filter":"envoy.filters.http.jwt_authn",
            "path": [
              {"key": "jwt_payload_okta_0"},
              {"key": "roles"}
            ],
            "value": {
              "stringMatch": {
                "exact": "testing"
              }
            }
          }
        }
      ]
    }
  }
}